docker-compose.yml

```diff
@@ -110,3 +155,33 @@ services:
     ports:
       - "4209:4209"
     command: mock docs/preservation-data-service.yml -p 4209 -h 0.0.0.0
+    healthcheck:
+      test: ["CMD-SHELL", "curl -f http://localhost:4209 || exit 1"]
+      interval: 10s
+      timeout: 5s
+      retries: 3
+
+  health_checker:
```

Comment:
If i understand correctly, this is a dummy container that only gets created once all other containers are healthy?That's convenient from the perspective of having `docker-compose` only exit once everything's ready -- but it also makes a user have to run a container that doesn't really do anything... maybe that's okay though since there are already a ton of containers here. Thoughts?

Reply:
Yeah, it's not an ideal arrangement. But if we didn't create a dummy container, we'd need to make each mockserver container depend on every other mockserver container which seems like a lot more duplicated config.  I'm open to ideas if you know of a better way to implement this healthcheck dependency scheme.

Reply:
It looks like we should be able to just make healthcheck a build command that doesn't do anything instead of a container: 

```
  health_checker:
    build: .
    command: [":"]
    depends_on:
      core:
        condition: service_healthy
```

Reply:
Seems like this instruction will build the image using the dockerfile at . and then run it. This works in our case because the prism container will just print out the helptext for the the `prism` CLI if there are no arguments specified in `command`.
```***:code42-mock-servers peter.briggs$ docker ps -a
CONTAINER ID   IMAGE                                 COMMAND                  CREATED         STATUS                        PORTS     NAMES
eb3eea41bfc8   code42-mock-servers_health_checker    "node dist/index.js :"   7 minutes ago   Exited (1) 53 seconds ago               code42-mock-servers_health_checker_1
0bdc36d58aae   c42/mock-microservice-endpoints:1.0   "node dist/index.js …"   7 minutes ago   Exited (137) 32 seconds ago             mock_preservation_data_service
...
```

Reply:
I think I might have an alternate solution here that kind of comes full circle:

https://github.com/vishnubob/wait-for-it
https://medium.com/@krishnaregmi/wait-for-it-docker-compose-f0bac30f3357

If we add the `wait-for-it` script to our repo, we can then create a startup script that first calls the same command that the existing Dockerfile does in its `ENTRYPOINT`, then calls the `wait-for-it` script. We can then override `ENTRYPOINT` in our own Dockerfile with a call to that script. This would make every container that is made from that dockerfile wait until its service is started before proceeding

https://github.com/stoplightio/prism/blob/master/Dockerfile

Reply:
surpise, the above doesn't work either -- it only works if you want to wait for a _different_ container, not the same one that this is running on. Talked with Peter offline and we agreed that while this feels funky, at this point we don't want to hold things up if we don't have any better ideas.

Reply:
Alan and I discussed this offline. The problem here is that the container lifecycle is tied to the originating process. So we can't have a docker startup script that a) instantiates a web server and then b) execute `wait-for-it.sh` to scan for a particular port on that web server to be available, and only mark the container as "started" once that port is available.

We agreed that although the current solution is less than ideal, since none of the suggested improvements have worked out that it isn't worth holding up this PR to implement.